### PR TITLE
Add update_security_access_file

### PR DESCRIPTION
**Pull request by bealdav**
WIP
- [X] create file security ir.model.access.xml for each model defined in module with base.user
- [ ] update existing file without add existing model
- [ ] allow to add several grants

### CODE REVIEW

File: update_security_access_file/README.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
+
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Update Security Access File
+===========================
+
+This module was written for **developpers** to easily generate and update
```
Comment (pedrobaeza) [inline update_security_access_file/README.rst:8]: s/developpers/developers

File: update_security_access_file/README.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+It search all the models defined by the module (except transient and abstract)
+and then produce a security file for those models.
+
+Usage
+=====
+
+To use this module, you need to:
+
+* go to modules list view.
+* choose a module and click on /!\\ image.
```
Comment (pedrobaeza) [inline update_security_access_file/README.rst:20]: Please insert the icon image here. Maybe another icon would be more aproppiated for this option.

File: update_security_access_file/README.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Update Security Access File
+===========================
+
+This module was written for **developpers** to easily generate and update
+model security files.
+
+It search all the models defined by the module (except transient and abstract)
```
Comment (pedrobaeza) [inline update_security_access_file/README.rst:11]: s/search/searches

**Comment by pedrobaeza** [comment]
I would call the module _security_access_file_exporter_

File: update_security_access_file/save_file.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                raise UserError(
+                    _("Impossible to update/create the file in this path '%s'")
+                    % security_path)
+        # TODO return info to the user
+        return True
+
+    @api.model
+    def add_missing_models(self, models, writer):
+        for model in models:
+            print model
```
Comment (pedrobaeza) [inline update_security_access_file/save_file.py:70]: Remove this line

File: update_security_access_file/save_file.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            data = []
+            for column in CSV_COLUMN:
+                data.append(struct[column])
+            writer.writerow(data)
+        return True
+
+    @api.model
+    def search_unaccess_models(self):
+        module_path = get_module_path(self.name)
+        print module_path
```
Comment (pedrobaeza) [inline update_security_access_file/save_file.py:97]: Remove this line

File: update_security_access_file/save_file.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+        classes_in_module = self.__metaclass__.module_to_models[self.name]
+        all_classes_created_by_the_module = (
+            [x for x in classes_in_module
+             # classes with _name attribute but not _inherit are required
+             if x._name and not hasattr(x, '_inherit')])
+        print 'all_classes_created_by_the_module', all_classes_created_by_the_module
```
Comment (pedrobaeza) [inline update_security_access_file/save_file.py:103]: Remove this line

File: update_security_access_file/save_file.py [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        model_names_security_required = (
+            [x._name for x in all_classes_created_by_the_module
+             # wizard classes shouldn't be include
+             if not x._transient and (
+                 # only concrete models 'Model' must be include
+                 # not 'AbstractModel'
+                 x._auto or x._table)])
+        print 'model_names_security_required', model_names_security_required
```
Comment (pedrobaeza) [inline update_security_access_file/save_file.py:111]: Remove this line

File: update_security_access_file/save_file.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        models = self.env['ir.model'].search(
+            [('model', 'in', model_names_security_required)])
+        print models
```
Comment (pedrobaeza) [inline update_security_access_file/save_file.py:114]: Remove this line

File: update_security_access_file/save_file.py [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        print [x.model for x in models]
```
Comment (pedrobaeza) [inline update_security_access_file/save_file.py:115]: Remove this line

**Comment by pedrobaeza** [comment]
You need to add unicodecsv dependency to .travis.yml to get a correct build on Travis

**Comment by pedrobaeza** [comment]
I have tried the module on runbot and see flaws:
- The icon is only presented on list view. It would be good to have it at least of form view (and even better on kanban).
- When you click it, it seems that nothing happens. I would expect that the security file would be presented for downloading. I suggested you the other name because I understood that this is the functionality.
- How do you define to a manual security rule that belongs to that module? Is the module limited to the pre-created security rules that maybe you have changed via UI?

**Comment by bealdav** [comment]
Hi Pedro, thanks for your review. 
I'll taken account of your remarks in next commit.

> Is the module is limited to the pre-created security rules

Yes, it's just to attach main grants to model of modules whether the UI actions.

For UI action to code, see #107

**Pull request closed without merging**